views/agency-requests/edit_participant.php

```diff
@@ -0,0 +1,345 @@
+<?php
+/**
+ * PEP Capping 2017 Algozzine's Class
+ *
+ * Allows user to edit participant information
+ *
+ * After a user selects a participant to view, they can 
+ * edit either the participant's information.
+ *
+ * @author Vallie Joseph
+ * @copyright 2017 Marist College
+ * @version 0.7.1
+ * @since 0.6.2
+ */
+
+include('header.php');
+global $db, $params ,$formid;
+?>
+
+<div class="d-flex flex-column w-100" style="height: fit-content;">
+<?php
+// Get people id from params
+$peopleid = rawurldecode(implode('/', $params));
+
+//grab participant info
+$db->prepare("get-participant-form", "SELECT * FROM people".
+			" INNER JOIN participants ON people.peopleid = participants.participantid ".
+			" LEFT JOIN forms ON participants.participantid = forms.participantid ".
+			" WHERE participants.participantid = $1 ");
+$result = $db->execute("get-participant-form", [$peopleid]);
+$participant = pg_fetch_assoc($result);
+
+extract($participant);
+$formid = $participant['formid'];
```

Comment:
If you're extracting $participant I don't think you need to do this. $formid will work without defining it first.

Reply:
Sorry a little confused on this comment, could you elaborate?

Reply:
`<?php

/* Suppose that $var_array is an array returned from
   wddx_deserialize */

$size = "large";
$var_array = array("color" => "blue",
                   "size"  => "medium",
                   "shape" => "sphere");
extract($var_array, EXTR_PREFIX_SAME, "wddx");

echo "$color, $size, $shape, $wddx_size\n";

?>`

As you can see in the example, when using $extract on an associative array you can simply call the key directly as a variable. So, if 'formid' exists within the $participant associative array, you can access it using $formid after saying extract(), without having to physically define it by setting it equal to $participant['formid']. You are essentially just re-defining it with the same data. If you try to get participant information anywhere else you can do the same. You just have to be careful of any clashing keys since you do it multiple times within the same file.

You can read about it here: http://php.net/manual/en/function.extract.php